plugins/modules/k8s_cluster_info.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Abhijeet Kasurde
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+module: k8s_cluster_info
+
+version_added: "0.11.1"
+
+short_description: Describe Kubernetes (K8s) API available and their respective versions
+
+author:
+    - Abhijeet Kasurde (@Akasurde)
+
+description:
+  - Use the OpenShift Python client to perform read operations on K8s objects.
+  - Authenticate using either a config file, certificates, password or token.
+  - Supports check mode.
+
+extends_documentation_fragment:
+  - community.kubernetes.k8s_auth_options
+
+requirements:
+  - "python >= 2.7"
+  - "openshift >= 0.6"
+  - "PyYAML >= 3.11"
+'''
+
+EXAMPLES = r'''
+- name: Get an existing Service object
+  community.kubernetes.k8s_cluster_info:
+  register: api_status
+'''
+
+RETURN = r'''
+apis:
```

Comment:
We should return some information in addition to the APIs. 

I'd like to see some information related to the `DynamicClient.configuration` object (an instance of the [kubernetes.client.Configuration](https://github.com/kubernetes-client/python/blob/master/kubernetes/client/configuration.py#L25)). We could pull the `host`, `api_key`, `username`, `password`, `verify_ssl`, `ssl_ca_cert`, `cert_file`, `key_file` and `proxy` (many of those would need to be `no_log`). This will allow the user to determine how the client is interacting with the cluster, even when the client was loaded via incluster config or kubeconfig (I've run into this need in the `okd` collection tests already)

I'd also like to see information about the cluster version (`DynamicClient.version`), and possibly about the client version as well (`openshift.__version__`), similarly to how `kubectl version` works.

Final output structure could look something like:

```
apis:
  # everything you have already
connection:
  host: 
  api_key:
  username:
  password:
  ...
version:
  server:
  client:
```